### ISSUE TITLE
DraggableScrollableActuator and GestureDetector is not working with showModalBottomSheet

### ISSUE THREAD

**Issue by vytautas-pranskunas-** [flutter/flutter#95680]
I am building application with bottom sheet and want to do some nice animations and hide some elements when bottom sheet is expanding but i cannot controll it. I tired out all possible ways of manipulating it and nothing works. 

I have tried put GestureDetector in multiple places (i thought maybe i could controll it through it but events are not fired)

Is there a way how to get gesture deltaY and apply it to sheet position?

```
class DraggableBottomSheet {
  static late BuildContext _draggableSheetContext;
  static late double _minExtent;
  static late double _maxExtent;
  static late void Function(void Function()) _setState;

  static bool _isExpanded = false;
  static late double _initialExtent;

  static Future<T> open<T>({
    required BuildContext context,
    required Widget child,
    Future<T> Function()? onCloseCallback,
    Future<T> Function()? onCloseButtonClick,
    Function(DragUpdateDetails)? onVerticalDragUpdate,
    bool isDismissible = true,
    Color backgroundColor = CustomColors.veryLightGrey,
    double initialChildSize = 0.4,
    double minChildSize = 0.1,
    double maxChildSize = 0.9,
    double horizontalPadding = 15,
  }) async {
    _minExtent = minChildSize;
    _maxExtent = maxChildSize;
    _initialExtent = initialChildSize;

    await showModalBottomSheet<T>(
        context: context,
        isScrollControlled: true,
        enableDrag: false,
        isDismissible: isDismissible,
        backgroundColor: backgroundColor,
        shape: const RoundedRectangleBorder(
          borderRadius: BorderRadius.vertical(top: Radius.circular(10)),
        ),
        builder: (BuildContext bc) {
          return WillPopScope(
            onWillPop: () => Future.value(isDismissible),
            child: GestureDetector(
                onTap: toggleSheet,
                onVerticalDragStart: (details) {
                  print('onVerticalDragStart');
                },
                onVerticalDragUpdate: (details) {
                  if (onVerticalDragUpdate != null) {
                    onVerticalDragUpdate(details);
                  }
                  print('onVerticalDragUpdate');
                },
                child: StatefulBuilder(builder: (context, setStateInner) {
                  _setState = setStateInner;
                  return DraggableScrollableActuator(
                    child: DraggableScrollableSheet(
                        initialChildSize: _initialExtent,
                        minChildSize: _minExtent,
                        maxChildSize: _maxExtent,
                        expand: false,
                        builder: (dContext, scrollController) {
                          _draggableSheetContext = dContext;

                          return SingleChildScrollView(
                            controller: scrollController,
                            child: Stack(clipBehavior: Clip.none, children: [
                              Column(
                                mainAxisAlignment: MainAxisAlignment.start,
                                children: <Widget>[
                                  const SizedBox(
                                      height: 30,
                                      child: Align(
                                          alignment: Alignment.topCenter,
                                          child: Icon(
                                            Icons.drag_handle,
                                            color: CustomColors.pageGrey300,
                                            size: 32,
                                          ))),
                                  Container(
                                      padding: EdgeInsets.symmetric(horizontal: horizontalPadding),
                                      width: MediaQuery.of(context).size.width,
                                      child: SafeArea(child: child))
                                ],
                              ),
                             
                            ]),
                          );
                        }),
                  );
                })),
          );
        });

    return onCloseCallback != null ? await onCloseCallback() : true as FutureOr<T>;
  }

  static void toggleSheet() {
    print('toggleSheet');
    _setState(() {
      _initialExtent = _isExpanded ? _minExtent : _maxExtent;
      _isExpanded = !_isExpanded;
    });
    DraggableScrollableActuator.reset(_draggableSheetContext);
  }
}

usage:

DraggableBottomSheet.open(
        context: context,
        child: ListView.builder(
                itemCount: 5,
                shrinkWrap: true,
                itemBuilder: (BuildContext context, int index) {
                  return Text('hello');
                },
              )),
        backgroundColor: Colors.white,
        horizontalPadding: 0,
        initialChildSize: 0.5,
        minChildSize: 0.5,
      );

toggle can be tested with:
DraggableBottomSheet.toggleSheet();

or pressing button inside panel
```

Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 2.5.3, on macOS 11.6.1 20G224 darwin-x64, locale
    en-GB)
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
[✓] Xcode - develop for iOS and macOS
[✓] Chrome - develop for the web
[✓] Android Studio (version 4.2)
[✓] VS Code (version 1.63.1)
[✓] Connected device (2 available)

• No issues found!

**Comment by mamuseferha** [comment]
Hi @vytautas-pranskunas- here's a discussion about draggableScrollablesheet https://github.com/flutter/flutter/issues/45009, have you also tried using a [Listener](https://api.flutter.dev/flutter/widgets/Listener-class.html) instead?

Let me know if any of these work for you.

**Comment by vytautas-pranskunas-** [comment]
@mamuseferha I was able to do it with NotificationListiner + rxdart. However changing bottom sheet position with DraggableScrollableActuator.reset and StateBuilder is not stright forward and causes a lot of headache because 
1) First i need to wrap everything in StatefulBuilder or user StatefullWidget
2) I need to update ```initialChildSize``` with setState and
3) I need to call DraggableScrollableActuator.reset

problems here
1) I cannot call DraggableScrollableActuator.reset stright after because setState need some time to rerender and set new ```initialChildSize``` so calling ```DraggableScrollableActuator.reset``` is not giving any effect as well as ```Future.delayed(const Duration.zero, () => DraggableScrollableActuator.reset(context))```. So I have to call it after some time - I am calling it after 100 ms...
2) Because of this from time to time I am getting strange error which I cannot figure out why:
```
flutter: #27     SchedulerBinding._handleDrawFrame
package:flutter/…/scheduler/binding.dart:996
flutter: #31     _invoke (dart:ui/hooks.dart:166:10)
flutter: #32     PlatformDispatcher._drawFrame (dart:ui/platform_dispatcher.dart:270:5)
flutter: #33     _drawFrame (dart:ui/hooks.dart:129:31)
flutter: (elided 3 frames from dart:async)
flutter: ════════════════════════════════════════════════════════════════════════════════════════════════════
flutter: Another exception was thrown: Null check operator used on a null value

```

3) No animations - if i call ```DraggableScrollableActuator.reset``` it jumops stright away to the point :(

Would you be able to help me to solve this error?

p.s. I feel like this should be more stright forward because now it feels like hacking...

**Comment by mamuseferha** [comment]
Hi @vytautas-pranskunas- 
Can you please provide a [minimal complete reproducible code sample](https://stackoverflow.com/help/minimal-reproducible-example) that captures these issues as well as your `flutter doctor -v`?
Thank you

**Comment by vytautas-pranskunas-** [comment]
@mamuseferha sure. Before that - would you tell me why ```DraggableScrollableController``` has not been implemented as done here: 
https://gist.github.com/caseycrogers/c972511cb8a9d0d29b594dc4bb3c1b35

**Comment by mamuseferha** [comment]
@vytautas-pranskunas- Looks like a pull request has been recently merged for this https://github.com/flutter/flutter/pull/92440 and introduces `animateTo` and `jumpTo`.

Please upgrade to the latest master channel to check this out, you can also refer to the pull request for more information and questions.

I'll go ahead and close this now. 

Thank you.

**Comment by vytautas-pranskunas-** [comment]
@mamuseferha thats why i had this question because I have upgradet to 2.8 and those commands are not there :( anyway I have solved my problem with 2.8 fix snap.
